aten/src/ATen/TensorImpl.h

```diff
@@ -65,6 +65,18 @@ struct TensorImpl : public Retainable {
     is_scalar = s;
   }
 
+  // True if a tensor was auto-wrapped from a C++ or Python number.
+  // Wrapped numbers are considered "lower priority" when computing the result
```

Comment:
Wrapped numbers do not participate in the result type calculation if there are any Tensors that are not wrapped numbers. (I'll improve the comment).

This is intended to keep backwards compatibility. We wrap all integers as 0-d LongTensors and Python floats as DoubleTensors, but we don't want to promote `byte_tensor + 5` to long or `float_tensor + 3.5` to double.

NumPy result_type calculation is a bit different. I'm not sure we want to use it because it promotes float arrays to double arrays in more cases than we would like.

Reply:
Makes sense. Still, we should take them into account when you do `byte_tensor + 2.5`, right?